fili-core/src/main/java/com/yahoo/bard/webservice/data/config/names/ApiMetricName.java

```diff
@@ -45,6 +51,23 @@ default boolean isValidFor(Granularity granularity) {
     }
 
     /**
+     * Determine if this API Metric Name is valid for the given time grain.
+     * This capability is provided as a convenience for configurers to filter metrics from a TableGroup onto a set of
+     * LogicalTable instances at configuration time.  This version allows the logical metric itself to be used for
+     * filtering.
+     * <p>
+     * An example of this is a DailyAverage metric than doesn't make sense to query by the HOUR grain, but does at WEEK.
+     *
+     * @param logicalMetric  The metric whose validity is being tested.
+     * @param granularity  TimeGrain to determine validity for
+     *
+     * @return True if the ApiMetricName is valid for the time grain
+     */
+    default boolean isValidFor(LogicalMetric logicalMetric, Granularity granularity) {
```

Comment:
Can we flip the order of these parameters? It's a little odd to have an overload that drops parameters from the front, rather than the back.